### ISSUE TITLE
Seafile dies when temp folder is full

### ISSUE THREAD

**Issue by denis-roy** [haiwen/seadrive-gui#30]
## VERSIONS

Seafile 6.0.9
Seadrive 0.7.0

## PROBLEM

I have add numerous problems (I/O errors) when I try to queue a lot of big files and seem to be running out of space at the same time on the drive that Seafile uses as a temporary folder, which is a small SSD drive.

I then have to restart the client for it to continue uploading.

## ASSUMPTIONS
My understanding is that when you copy a file to Seadrive, it gets moved to a temporary folder where it is queued for upload `(C:\Users\{USER}\seadrive\data\file-cache\)`?

I think these kinds of errors should be handled as exceptions so as to provide the user with a relevant error message and I also would like to be able to define my own temporary folder for Seafile from within the Advanced Settings tab of the app.

Otherwise, I believe the drive is great. Especially for files that are nice to have on-demand but not necessarily synched like music/movies/software installers. Keep up the good work and thanks to all contributors :v:

**Comment by monkeydri** [comment]
Same error here, seadrive crashed after hitting the 10GB cache limit, now the data that I transfered into the Seadrive library is gone (well it is somewhere inside file-cache but I could not find a way to read it...).

STR :

- transfer big files to a Seadrive library (folder inside seadrive like `SeaDrive/My Libraries/some-folder`)
- wait for the transfer to complete
- wait for the limit to be reached => Seadrive crashes
- start Seadrive again.
- remaining files are not in the original location (they have been moved to Seadrive cache) neither in Seadrive folder.

This should be high priority as I can lead to data loss.
